src/main/java/uk/nhs/hee/tis/trainee/credentials/event/PlacementEventListener.java

```diff
@@ -61,7 +64,21 @@ void deletePlacement(DeleteEventDto deletedPlacement) {
   @SqsListener("${application.aws.sqs.update-placement}")
   void updatePlacement(UpdateEventDto updatedPlacement) {
     log.info("Received update event for placement {}.", updatedPlacement);
-    // For now, we simply revoke regardless of which fields have updated (pending TIS21-4152)
-    revocationService.revoke(updatedPlacement.tisId(), CredentialType.TRAINING_PLACEMENT);
+
+    RecordDto recrd = updatedPlacement.recrd();
+    String specialty = recrd.getData().get("TPL-Specialty");
+    String grade = recrd.getData().get("TPL-Grade");
+    String nationalPostNumber = recrd.getData().get("TPL-PlacementNPN");
+    String employingBody = recrd.getData().get("TPL-EmployingBodyOfPost");
+    String site = recrd.getData().get("TPL-Site");
+    String startDate = String.valueOf(
+        LocalDate.parse(recrd.getData().get("TPL-PlacementStartDate")));
+    String endDate = String.valueOf(LocalDate.parse(recrd.getData().get("TPL-PlacementEndDate")));
```

Comment:
I don't think this UpdateEvent will have these attribute names, those are only used in the credential itself.